credentials/src/main/kotlin/web5/sdk/credentials/VerifiableCredential.kt

```diff
@@ -306,7 +309,108 @@ public class VerifiableCredential internal constructor(public val vcDataModel: V
      * ```
      */
     public fun fromJson(vcJson: String): VerifiableCredential {
-      return VerifiableCredential(VcDataModel.fromJson(vcJson))
+      val typeRef = object : TypeReference<HashMap<String, Any>>() {}
+      val vcMap = objectMapper.readValue(vcJson, typeRef)
+      validateDataModel(vcMap)
+      return VerifiableCredential(VcDataModel.fromMap(vcMap))
+    }
+
+    private fun validateDataModel(model: Map<String, Any>) {
```

Comment:
Should we be using the `@Throws` annotation here? Seems like there's two possible outcomes for this function:
1. The function completes
2. The function throws

That didn't really occur to me on my first read-through that it'd throw. Would be great to add for clarity for readers!

Reply:
Excellent point, added comments. Let me know if it's more clear now.

Reply:
This could live in a new validators file

Reply:
I would rather keep it private unless it's necessary to make this function public. Do you have a use-case in mind that requires it to be public?